csrc/ir/utils.h

```diff
@@ -427,16 +427,34 @@ void validateDomainEquivalence(
     const std::vector<IterDomain*>& derived_domain);
 
 //! Check if all the inputs required to compute needed_val are known
-bool dependenciesSatisfied(
-    std::vector<const Val*> needed_vals,
-    std::unordered_set<const Val*> known_vals = {});
-
+template <
+    typename ValOrVectorOfVal,
+    typename SetOfVal = std::unordered_set<const Val*>>
 inline bool dependenciesSatisfied(
-    std::vector<Val*> needed_vals,
-    std::unordered_set<Val*> known_vals = {}) {
-  return dependenciesSatisfied(
-      std::vector<const Val*>(needed_vals.begin(), needed_vals.end()),
-      std::unordered_set<const Val*>(known_vals.begin(), known_vals.end()));
+    ValOrVectorOfVal needed_vals,
+    const SetOfVal& known_vals = {}) {
+  if constexpr (
+      std::is_same_v<ValOrVectorOfVal, const Val*> ||
+      std::is_same_v<ValOrVectorOfVal, Val*>) {
+    return dependenciesSatisfied(
+        std::vector<const Val*>{needed_vals}, known_vals);
+  } else {
```

Comment:
can you add some comments? e.g. convert single val* to vector, default container is std::unordered_set<const Val*>>, what happens if other container is used, e.g. std::unordered_map.

Reply:
added